File: comp.py
import logging
import sys

logging.basicConfig(filename="/dev/stdout", level=logging.DEBUG, format="%(message)s")

from ebb.interp import Env

def read(name):
    with open(name) as f:
        return f.read()

env = Env()

env.source(read("ebb.ebb"))
env.call("main-lang", (sys.argv[1], read(sys.argv[2])))

File: driver/comp.lua

package.path = package.path .. ';temper.out/lua/?.lua;temper.out/lua/?/init.lua'

local interp = require('eb.interp')
local value = require('eb.value')

local function read(name)
    local f = io.open(name)
    local r = f:read('*all')
    f:close()
    return r
end

local env = interp.Env()
local ebfile = read('eb/eb.eb')
env:source(ebfile)
env:call('main-lang', {value.string('c'), value.string(arg[1])})
env:flush()
